tesseract_python/tesseract_python/CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+
+  set(CMAKE_SWIG_OUTDIR ${CMAKE_CURRENT_BINARY_DIR}/python/tesseract_ros)
+  set(SWIG_OUTFILE_DIR ${CMAKE_CURRENT_BINARY_DIR})
+  swig_add_module(tesseract_ros_python python swig/ros/tesseract_ros_python.i)
+  swig_link_libraries(tesseract_ros_python 
+    ${tesseract_rosutils_LIBRARIES}
+    ${tesseract_msgs_LIBRARIES}
+    tesseract::tesseract_motion_planners_trajopt 
+    tesseract::tesseract_motion_planners_core 
+    tesseract::tesseract_visualization 
+    tesseract::tesseract trajopt::trajopt 
+    ${PYTHON_LIBRARIES} 
+    ${EIGEN3_LIBRARIES}
+	${roscpp_LIBRARIES}
+    ${roslib_LIBRARIES}
+	${rosconsole_LIBRARIES}
+	)
+	
+  target_include_directories(${SWIG_MODULE_tesseract_ros_python_REAL_NAME} PUBLIC 
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/swig>"
+      "$<INSTALL_INTERFACE:include>")
+  target_include_directories(${SWIG_MODULE_tesseract_ros_python_REAL_NAME}  SYSTEM PUBLIC
+      ${tesseract_rosutils_INCLUDE_DIRS}
+      ${EIGEN3_INCLUDE_DIRS}
+      ${console_bridge_INCLUDE_DIRS}
+      ${Boost_INCLUDE_DIRS}
+      ${TinyXML2_INCLUDE_DIRS}
+	  ${PYTHON_INCLUDE_DIRS}
+	  ${NUMPY_INCLUDE_DIR}
+      $<TARGET_PROPERTY:tesseract::tesseract,INTERFACE_INCLUDE_DIRECTORIES>)
+
+  set_target_properties(${SWIG_MODULE_tesseract_ros_python_REAL_NAME}
+    PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/python/tesseract_ros)
+
+  add_custom_command(TARGET ${SWIG_MODULE_tesseract_ros_python_REAL_NAME} POST_BUILD COMMAND 
+    ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/swig/ros/__init__.py
+    $<SHELL_PATH:$<TARGET_FILE_DIR:${SWIG_MODULE_tesseract_ros_python_REAL_NAME}>>/__init__.py)
+
+endif()
+
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake/setup.py.in" "${CMAKE_CURRENT_BINARY_DIR}/python/setup.py" @ONLY)
+
+set(enable_setuptools_deb_layout OFF)
+if(EXISTS "/etc/debian_version")
+  set(enable_setuptools_deb_layout ON)
+endif()
+option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
+if(SETUPTOOLS_DEB_LAYOUT)
+  message(STATUS "Using Debian Python package layout")
+  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
+endif()
+
+install(CODE "message(STATUS \"Running setup.py in ${CMAKE_CURRENT_BINARY_DIR}/python\")
+execute_process(COMMAND python setup.py install -f
+--root=/ --prefix=${CMAKE_INSTALL_PREFIX} ${SETUPTOOLS_ARG_EXTRA} --single-version-externally-managed WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/python)")
+
+if(TESSERACT_ENABLE_TESTING)
+  enable_testing()
+
+  tesseract_add_run_tests_target()
+
+  add_test (NAME python-tests
+  COMMAND ${PYTHON_EXECUTABLE} -m pytest -p no:cacheprovider
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  
+  find_package(tesseract_support REQUIRED)
+  if (WIN32)
+  set_tests_properties(python-tests PROPERTIES ENVIRONMENT "TESSERACT_SUPPORT_DIR=${TESSERACT_SUPPORT_DIR}")
```

Comment:
Could this be extracted from the target INTERFACE_COMPILE_DEFINITIONS?

```
set_target_properties(tesseract::tesseract_support PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "TESSERACT_SUPPORT_DIR=\"/home/larmstrong/catkin_ws/trajopt_only_ws/install/share/tesseract_support\""
)
```

Reply:
Possibly, but it would require parsing INTERFACE_COMPILE_DEFINITIONS to extract that value